src/Illuminate/Foundation/Testing/TestCase.php

```diff
@@ -143,6 +143,14 @@ protected function setUpTraits()
             $this->setUpFaker();
         }
 
+        foreach ($uses as $trait) {
+            $method = 'boot'.class_basename($trait);
```

Comment:
I’ve implemented something similar and I love this! Only critique: since this is getting called from the `setUp` method I feel like it should call `setUpTraitName`. Boot has a pretty specific meaning within the application lifecycle and this actually fires after the booting is done.